crates/ethrpc/src/current_block/mod.rs

```diff
@@ -130,6 +128,41 @@ pub async fn current_block_stream(
     Ok(receiver)
 }
 
+/// Returns a stream that is synchronized to the passed in stream by only yields
+/// every nth update of the original stream.
```

Comment:
Maybe it makes sense to explicitly mention that this type of stream can't be used for catching reorgs/any other parent-block-related logic.